projects/plugins/jetpack/_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -3027,14 +3093,21 @@ public static function validate_non_neg_int( $value, $request, $param ) {
 	public static function validate_list_item( $value, $request, $param ) {
 		$attributes = $request->get_attributes();
 		if ( ! isset( $attributes['args'][ $param ] ) || ! is_array( $attributes['args'][ $param ] ) ) {
-			return new WP_Error( 'invalid_param', sprintf( esc_html__( '%s not recognized', 'jetpack' ), $param ) );
+			return new WP_Error(
+				'invalid_param',
+				sprintf(
+					/* Translators: Placeholder is a parameter name. */
+					esc_html__( '%s not recognized', 'jetpack' ),
+					$param
+				)
+			);
 		}
 		$args = $attributes['args'][ $param ];
 		if ( ! empty( $args['enum'] ) ) {
 
 			// If it's an associative array, use the keys to check that the value is among those admitted.
 			$enum = ( count( array_filter( array_keys( $args['enum'] ), 'is_string' ) ) > 0 ) ? array_keys( $args['enum'] ) : $args['enum'];
-			if ( ! in_array( $value, $enum ) ) {
+			if ( ! in_array( $value, $enum, true ) ) {
```

Comment:
Let's hope no one is passing ints in `$args['enum']` or using arrays with some string and some numeric keys. At a quick grep I didn't see any such uses.

I suppose we could do a `$enum = array_map( 'strval', $enum );` if we wanted to account for the possibility.

Reply:
👍 a4659fc099ba7e17d4affe4629c713760c2b7618